### PR TITLE
Feat(eos_designs): Add RD and RT override for VRFs

### LINKED ISSUE #3296: It is better to have a possibility to customize the RD/RT of VRF
### Enhancement summary

For now, the RT/RD value can only be generated by vrf_id or vrf_vni. But in some customers' env, they normally use many different values, not just <vrf_id>:<vrf_id>. In case for the inter-communication with existing un-ruled vrfs, a flexible way to tune this value is necessary.

### Which component of AVD is impacted

eos_designs, eos_cli_config_gen

### Use case example

For example, VRF xxx RD is VTEP IP:1001, RT evpn for both 1001:1 (1001 is an id we defined, not the vrf_id or vni). 

### Describe the solution you would like

Like mac-vrf using rd/rt_override. These 2 vars could help to generate any kinds of RD/RT values.

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by bjmeuer**
## Change Summary

Added the possibility to override the RD and RT values for VRFs. 

## Related Issue(s)

Fixes #3296

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
Added RD and RT override for VRFs in similar way it was already available for l2vlans

```
tenants:
  - name: TENANT1
    mac_vrf_vni_base: 10000
    vrfs:
      - name: VRF1
        vrf_id: 1
        rd_override: 111:222
        rt_override: 333:444
        ...
      - name: VRF2
        vrf_id: 2
        rd_override: 999
        rt_override: 999
        ...
```

## How to test
```
tenants:
  - name: TENANT1
    mac_vrf_vni_base: 10000
    vrfs:
      - name: VRF1
        vrf_id: 1
        rd_override: 111:222
        rt_override: 333:444
        ...
      - name: VRF2
        vrf_id: 2
        rd_override: 999
        rt_override: 999
        ...
```

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by ClausHolbechArista — APPROVED** [review]

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_network_services.schema.yml [hunk 1]
```diff
@@ -289,6 +289,28 @@ $defs:
                   "vrf_id" is used as default value for "vrf_vni" and "ospf.process_id" unless those are set.
                   "vrf_id" may also be used for VRF RD/RT ID. See "overlay_rd_type" and "overlay_rt_type" for details.
                   "vrf_id" is preferred over "vrf_vni" for MLAG iBGP peering vlan, see "mlag_ibgp_peering_vrfs.base_vlan" for details.
+              rd_override:
+                type: str
+                convert_types:
+                  - int
+                description: |
+                  By default the VRF RD will be derived from overlay loopback and vrf_id.
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_network_services.schema.yml]:
```suggestion
                  By default, the VRF RD will be derived from the pattern defined in `overlay_rd_type`.
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_network_services.schema.yml [hunk 2]
```diff
@@ -289,6 +289,28 @@ $defs:
...
+
+                  rd_override supports two formats:
+                    - A single number which will be used in the RD assigned number field instead of vrf_id (see 'overlay_rd_type' for details).
+                    - A full RD string with colon seperator which will override the full RD.
+              rt_override:
+                type: str
+                convert_types:
+                  - int
+                description: |
+                  By default the VRF RT will be derived from the vrf_id.
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_network_services.schema.yml]:
```suggestion
                  By default, the VRF RD will be derived from the pattern defined in `overlay_rt_type`.
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_network_services.schema.yml [hunk 3]
```diff
@@ -289,6 +289,28 @@ $defs:
...
+              rd_override:
+                type: str
+                convert_types:
+                  - int
+                description: |
+                  By default the VRF RD will be derived from overlay loopback and vrf_id.
+                  The rd_override allows us to override this value and statically define it.
+
+                  rd_override supports two formats:
+                    - A single number which will be used in the RD assigned number field instead of vrf_id (see 'overlay_rd_type' for details).
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_network_services.schema.yml]:
```suggestion
                    - A single number will be used in the RD assigned number subfield (second part of the RD).
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_network_services.schema.yml [hunk 4]
```diff
@@ -289,6 +289,28 @@ $defs:
...
+              rt_override:
+                type: str
+                convert_types:
+                  - int
+                description: |
+                  By default the VRF RT will be derived from the vrf_id.
+                  The rt_override allows us to override this value and statically define it.
+
+                  rt_override supports two formats:
+                    - A single number which will be used in the RT fields instead of vrf_id (see 'overlay_rt_type' for details).
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_network_services.schema.yml]:
```suggestion
                    - A single number will be used in the RT assigned number subfield (second part of the RT).
```

**Review by philippebureau — APPROVED** [review]
Tested, LGTM
I would commit Carl`s suggestions on the documentation

**Review by carlbuchmann — CHANGES_REQUESTED** [review]
LGTM, small suggestion on the documentation

**Comment by bjmeuer** [comment]
> LGTM, small suggestion on the documentation

Thanks for the review, I commited your suggestions

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Comment by forcecity** [comment]
I have 2 points to double check with you:
1. You have added rt_override, but for the logic of it, I’m just a little bit confused from your source code (router_bgp.py):
`def get_vrf_rt(self, vrf: dict) -> str:
        """
        Return a string with the route-target for one VRF
        """
        rt_override = default(vrf.get("rt_override"))

        if ":" in str(rt_override):
            return rt_override

        if rt_override is not None:
            return f"{rt_override}:{rt_override}"
        elif self._vrf_rt_admin_subfield is not None:
            admin_subfield = self._vrf_rt_admin_subfield
        elif self.shared_utils.overlay_rt_type["vrf_admin_subfield"] == "vrf_vni":
            admin_subfield = self.get_vrf_vni(vrf)
        else:
            # Both for 'id' and 'vrf_id' options.
            admin_subfield = self.get_vrf_id(vrf)
        return f"{admin_subfield}:{self.get_vrf_id(vrf)}"
`
On this line: return f"{rt_override}:{rt_override}", would it be return f”{ admin_subfield }:{ rt_override }”? Or you could define a new var like “assigned_number_subfield” and assign the rt_override value to it.

If I understand correctly from the eos_designs.schema, A single number will be used in the RT assigned number subfield (second part of the RT). But if you specify like the yellow line, the result will be different. By default, the “admin_subfield” will be vrf_id, and if I set different vrf_id and rt_override. In our deployment, it should be OK, but other customers may confuse about this behavior. What do you think?
2. I think there is a typo in schema (arista/avd/roles/eos_designs/schemas/schema_fragments/defs_network_services.schema.yml), line 308, it should be "the VRF **RT** will be derived"

**Comment by ClausHolbechArista** [comment]
@forcecity very good catch! We should also be respecting the admin_subfield setting if you give rt_override (without colon). I will fix this in a new PR.

**Pull request merged**